player.js

```diff
@@ -27,39 +30,69 @@ const getRandomPath = array => {
 const allContents = JSON.parse(fs.readFileSync('./.data/allContents.json'))
 const playlist = []
 
-const mplayerOptions = { verbose: !!(argv.verbose || argv.v), debug: !!(argv.debug || argv.d) }
+const mplayerOptions = {
+  verbose: !!(argv.verbose || argv.v),
+  debug: !!(argv.debug || argv.d),
+  args: '-ass -vo gl_nosw -fixed-vo',
+}
 if (argv.novideo) {
-  mplayerOptions.args = '-vo null'
+  mplayerOptions.args += ' -vo null'
 }
 const mplayer = new Mplayer(mplayerOptions)
 
-
-const express = require('express')
-const app = express()
-const startPlayer = files => {
-  mplayer.openFile(getRandomPath(files))
-  mplayer.on('stop', () => mplayer.openFile(getRandomPath(files)))
+const startRandomPlayer = (player, files) => {
+  player.openFile(getRandomPath(files))
+  player.on('stop', () => player.openFile(getRandomPath(files)))
 }
 
-app.get('/contents', (req, res) => res.json(allContents))
+if (argv.random) {
+  startRandomPlayer(mplayer, allContents)
+} else {
+  const app = express()
+  app.use(bodyParser.json())
 
-app.post('/request', (req) => {
-  const id = parseInt(req.body.id, 10)
-  const content = allContents.find(c => c.id === id)
-  if (!content) return // res json error
-  const existingContent = playlist.find(c => c.id === id)
-  if (existingContent) return // res json error 2
-  playlist.push(content)
-})
+  const startPlayer = (player, files) => {
+    const firstKara = files.shift()
+    console.log(firstKara.path)
+    player.openFile(firstKara.path)
+    player.on('stop', () => {
+      if (files.length) {
+        const kara = files.shift()
+        console.log(kara.path)
+        player.openFile(kara.path)
+      } else {
```

Comment:
Hence the files.shift() that depop the array yes, it doesn't loop and the fullscreen is a toggle. When the end is reached, the player show that it's the end (otherwise, it just stays idle with a black screen)